include/zephyr/logging/log_core.h

```diff
@@ -222,7 +222,7 @@ static inline char z_log_minimal_level_to_char(int level)
 		break; \
 	} \
 	/* For instance logging check instance specific static level */ \
-	if (_inst & !IS_ENABLED(CONFIG_LOG_RUNTIME_FILTERING)) { \
+	if (_inst && !IS_ENABLED(CONFIG_LOG_RUNTIME_FILTERING)) { \
```

Comment:
Since we are talking about the static analysers, we might as well also fix the [Rule 14.4](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#coding-guidelines) violation here:

```suggestion
	if (_inst != 0 && !IS_ENABLED(CONFIG_LOG_RUNTIME_FILTERING)) { \
```

Reply:
@stephanosio fair enough, fixed, thanks.